File: python/invertBinaryTree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None) -> None:
        self.val = val
        self.left = left
        self.right = right


def invertTree(root: TreeNode | None) -> TreeNode | None:
    return root


# Time -> 0(N) Space -> 0(log(N))
def recursiveSwap(root: TreeNode | None):
    if root is None:
        return

    root.left, root.right = root.right, root.left

    recursiveSwap(root.left)
    recursiveSwap(root.right)

File: go/InvertBinaryTree/InvertBinaryTree.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func invertTree(root *TreeNode) *TreeNode {
	recursiveSwap(root)
	return root
}

// Time -> 0(N) Space -> 0(log(N))
func recursiveSwap(root *TreeNode) {
	if root == nil {
		return
	}

	root.Left, root.Right = root.Right, root.Left
	recursiveSwap(root.Left)
	recursiveSwap(root.Right)
}
